### ISSUE TITLE
App Hanging, and then Cache is lost after clearing RAM 

### ISSUE THREAD

**Issue by ajk-47** [25huizengek1/ViTune#424]
### Steps to reproduce the bug

Sometimes the app displays "The Returned video Id doesn't match the requested one." This occurs both with my wifi and data connection and with no VPN intermediary, which leads me to believe that it is not due to the connection to the server or an IP ban, but a bug. Then, I either spam songs until it crashes, clear my RAM, or "kill app internals". Either way, it then ends up working just fine, but each song takes a little while to load. The most recent time this has happened, I checked my cache before clearing anything, and it was at 171MB, and afterwards it was back to 0MB. I did not screenshot it, but I have attached a screenshot of it at 61MB~ to show that it went down from my previous high of 171. (I listened to more music before bug reporting)

### Expected behavior

The App should cache songs, not request them from the server again, and just olay them "offline". The Cache should remain intact. (Or is it programmed to reset the cache?)

### Actual behavior

The app requested information from the server, the information didn't match the cache, and then the song refused to play and the cache was lost. (Or something.)

### Screenshots/Screen recordings

I am doing this on my phone and am scared to lose my writing. I will attach a photo after submitting the report. Thank you for looking into my issue with your limited time, I really appreciate that!

### Logs

I do not have access to the logs currently, but I would be happy to set up adb over the weekend/do something else to be of help. 

### ViTune version

v1.0.8

### What kind of build are you using?

Release (GitHub / F-Droid)

### Android version

Android 13

### Device info

I have a Motorola 5g 2022. It doesn't have a cache clearing function like Samsungs do and I do not have any such app that would automatically clear the cache from ViTune. 

### Upstream reproducibility

This doesn't occur in ViMusic.

### Checklist

- [X] I am able to reach YouTube Music through the **official** app/website and can confirm that I live in a [supported country](https://support.google.com/youtubemusic/answer/6313540?hl=en) (and this country is not temporarily unavailable)
- [X] I am not using a proxy, VPN, device policy or anything else that restricts access to YouTube Music
- [X] This issue has not been submitted already (this is not a duplicate of another issue)

**Comment by ajk-47** [comment]
I know this is a repeatish bug, but I also included the info about the Cache being lost so I made one anyways.
![Screenshot_20240911-191037](https://github.com/user-attachments/assets/d10ef984-3282-4d09-a477-170aabb30941)
![Screenshot_20240806-150253](https://github.com/user-attachments/assets/050a0fa7-99a3-4ff3-b736-1f907e156b6c)
and then the same song working in ViMusic and on the same internet connection: 
![Screenshot_20240806-150309](https://github.com/user-attachments/assets/73f37e86-c357-4ab3-9689-8fc783d62beb)

**Comment by 25huizengek1** [comment]
@ajk-47 Thank you so much for your bug report, it is very detailed so I think we can sort this out.

Could you please update your ViTune first and try again with the latest version (v1.0.8). I'm sure the screenshot isn't ViTune v1.0.8, because the error says "video **id**" instead of "video **ID**", and that change was introduced in v1.0.7.

**Comment by 25huizengek1** [comment]
Also, ViTune relies on Piped for extracting the stream urls from YouTube, and I've been putting off getting rid of this dependency on Piped for a while now, so maybe this is a good reason to do so, since Piped availability/downtime cause a lot of these issues currently.

**Comment by ajk-47** [comment]
This happened on v1.0.8. I just was in the car Wednesday when it happened, and I knew I had screenshotted the same issue in v1.0.7, so I didn't screenshot the error again. I use obtanium to stay upfate, so I know my app was up to date. But good eye, yeah that screenshot is from v1.0.7.

**Comment by 25huizengek1** [comment]
Should be fixed by 531b531aeb1b4529d934dc6b54af1157ca08a90f

**Comment by ajk-47** [comment]
I actually figured out that piped is blocked by at&t. So that explains part
of why my app was displaying errors
